Automattic/wp-calypso
# Disable Titan Control Panel via manage mailboxes on mobile

#### Changes proposed in this Pull Request

* The Titan Control Panel has blocked mobile access on the Control Panel. Until Titan Control Panel adds support for mobile devices, the Titan Manage Mailbox page should also not allow the same from opening on mobile devices.
* Changed the Titan Manage items to a render from an array/config object for abstraction and easier handling of the `VerticalNavItemEnhanced` component's `prop`s.

##### Caveats
* Sending `path` as `""` to `VerticalNavItemEnhanced` would only open the same URL in the new tab. Therefore, I created a new method, `getPathObject` to allow passing down props without the `path` and `description` prop in the `VerticalNavItemEnhanced` component.

#### Minor enhancement on component `VerticalNavItemEnhanced`
* Added support for passing down `className` as a prop to `VerticalNavItemEnhanced`

#### Testing instructions



Before:
![image](https://user-images.githubusercontent.com/1598017/119627896-3c081a80-be2a-11eb-98a5-af5ef938c9ba.png)

After:
![image](https://user-images.githubusercontent.com/1598017/119630814-eed97800-be2c-11eb-847d-d866b1435a4a.png)

Use the following instructions to switch to the feature branch to test this:
```shell
git remote add titan git@github.com:talk-to/wp-calypso.git
git fetch titan -p
git checkout add/disable-titan-manage-mailbox-mobile
```

* The branch `add/disable-titan-manage-mailbox-mobile` is available on Titan's fork - https://github.com/talk-to/wp-calypso/tree/add/disable-titan-manage-mailbox-mobile
* On your localhost, please go to `Upgrades -> Emails` and click on any WordPress (Titan) email.
* Click on `Manage all mailboxes`
* Open the DevTools and switch to `Responsive Mode` of development.
* Reload the page
* You should see all the links to the Titan Control Panel as disabled, except the `Get mobile app` link.

cc: @daledupreez
